cashctrl_ledger/accounts.py

```diff
@@ -1,166 +1,118 @@
+            if "tax_code" in incoming.columns:
+                payload["taxId"] = None if pd.isna(row["tax_code"]) else \
+                    self._client.tax_code_to_id(row["tax_code"])
+            self._client.post("account/update.json", data=payload)
+        self._client.invalidate_accounts_cache()
 
-    # Update account categories
-    def get_nodes_list(path: str) -> List[str]:
-        parts = path.strip("/").split("/")
-        return ["/" + "/".join(parts[:i]) for i in range(1, len(parts) + 1)]
-
-    def account_groups(df: pd.DataFrame) -> Dict[str, str]:
-        if df is None or df.empty:
-            return {}
-
-        df = df.copy()
-        df["nodes"] = [
-            pd.DataFrame({"items": get_nodes_list(path)}) for path in df["group"]
-        ]
-        df = unnest(df, key="nodes")
-        return df.groupby("items")["account"].agg("min").to_dict()
-
-    self._client.update_categories(
-        resource="account",
-        target=account_groups(target),
-        delete=delete,
-        ignore_account_root_nodes=True,
-    )
-    super().mirror_accounts(target, delete)
-
-def _single_account_balance(
-    self, account: int, date: Union[datetime.date, None] = None
-) -> dict:
-    """Calculate the balance of a single account in both account currency
-    and reporting currency.
-
-    Args:
-        account (int): The account number.
-        date (datetime.date, optional): The date for the balance. Defaults to None,
-            in which case the balance on the last day of the current fiscal period is returned.
-
-    Returns:
-        dict: A dictionary with the balance in the account currency and the reporting currency.
-    """
-    account_id = self._client.account_to_id(account)
-    params = {"id": account_id, "date": date}
-    response = self._client.request("GET", "account/balance", params=params)
-    balance = float(response.text)
-
-    account_currency = self._client.account_to_currency(account)
-    if self.reporting_currency == account_currency:
-        reporting_currency_balance = balance
-    else:
-        response = self._client.get(
-            "fiscalperiod/exchangediff.json", params={"date": date}
+    def delete(self, id: pd.DataFrame, allow_missing: bool = False) -> None:
+        incoming = enforce_schema(pd.DataFrame(id), self._schema.query("id"))
+        ids = []
+        for account in incoming["account"]:
+            id = self._client.account_to_id(account, allow_missing)
+            if id is not None:
+                ids.append(str(id))
+        if len(ids):
+            self._client.post("account/delete.json", {"ids": ", ".join(ids)})
+            self._client.invalidate_accounts_cache()
+
+    def mirror(self, target: pd.DataFrame, delete: bool = False):
+        """Synchronizes remote CashCtrl accounts with a desired target state
+        provided as a DataFrame.
+
+        Updates existing categories before creating accounts and then invokes
+        the parent class method.
```

Comment:
Thanks for the clear description. Makes it immediately clear why we need a specific mirror method here.

Also, thanks for the efficient and architecture: First take preparation steps specific to CashCtrl, then invoke the parent class method.